src/PluginBase.php

```diff
@@ -24,6 +24,18 @@ abstract class PluginBase implements PluginInterface, EventSubscriberInterface
 
 	protected $io;
 
+	protected $configPath;
+
+	protected $config;
+
+	private static $defaultConfig = [
+		'keyDirectory'     => '',
+		'package'          => '',
+		'pharName'         => '',
```

Comment:
We would need multiple entries for pharName,Url and signature for cases where multiple phars shall be installed (think phpcs and phpcbf)

Reply:
So `mediator.json` would look like this
```javascript
{
	"packageName": "foo/bar",
	"keyDirectory": "keys",
	"phars": [
		{
			"pharName": "foo",
			"pharUrl": "...",
			"pharSignatureUrl": "..."
		}
	]
}
```

Reply:
Makes sense. And maybe it's overkill, but how about creating proper objects to encapsulate the values rather than having an ugly nested array?

Reply:
Yes after agreeing how the configuration should look we should definitely do that :)